app/controllers/portal/client_logins_controller.rb

```diff
@@ -27,15 +26,32 @@ def create
             locale: I18n.locale,
             visitor_id: visitor_id
           )
-          session[:sms_phone_number] = @form.sms_phone_number
         end
 
-        redirect_to login_link_sent_portal_client_logins_path
+        @verification_code_form = Portal::VerificationCodeForm.new(contact_address: @form.email_address || local_phone_number(@form.sms_phone_number))
+        render :enter_verification_code
       else
         render :new
       end
     end
 
+    def check_verification_code
+      contact_method = params.require(:portal_verification_code_form).permit(:contact_method)
+      verification_code = params.require(:portal_verification_code_form).permit(:verification_code)
+      hashed_verification_code = VerificationCodeService.digest(contact_method, verification_code)
+      # Remove if-production check when lockout counters are implemented
+      if Rails.env.production?
+        head status: 403
+        return
+      end
+
+      if ClientLoginsService.clients_for_token(hashed_verification_code).exists?
```

Comment:
- [ ] Make a method in `VerificationCodeService` called `clients_for_token` which takes contact_method + verification_code => set of Clients; this should validate that the clients are not locked-out too
- [ ] In the controller, if that set is empty, use `by_contact_info` to increment lockout